### PR TITLE
Update docs and release notes RE TOWER_ENABLE_ARM64

### PR DESCRIPTION
**Pull request by llewellyn-sl**
Per https://seqera.slack.com/archives/C03KZDQ0UL9/p1715279417161679, the env var to enable ARM64 CPU support is not documented in version 23.2 and 23.3 docs. This PR includes:

- Adding TOWER_ENABLE_ARM64 to the configuration overview for 23.2 and 23.3
- Correcting changelog and release note entries for Graviton support
- Noting the config requirement in the 23.2 and 23.3 versions of the AWS Batch CE user guide page
- Noting the removal of the config variable in 23.4 release notes

### CODE REVIEW

File: platform_versioned_docs/version-23.2.0/compute-envs/aws-batch.mdx [hunk 1]
```diff
@@ -172,14 +172,13 @@ The bucket should be in the same region selected in the previous step.
...
 
-:::note
-Requires Fargate, Wave containers, and Fusion v2 file system to be enabled. This feature is not compatible with GPU-based architecture.
-
-        This is a preview feature that requires the use of Nextflow `23.06.0-edge`. Graviton support is currently only available in a [Tower Cloud](https://tower.nf/) public instance.
+    :::note
+    Requires Fargate, Wave containers, and Fusion v2 file system to be enabled. This feature is not compatible with GPU-based architecture.
 
-:::
+    This is a preview feature that requires the use of Nextflow `23.06.0-edge`. In Tower Enterprise deployments, set `TOWER_ENABLE_ARM64=true` to enable Graviton instance support.
```
Comment (gwright99) [inline platform_versioned_docs/version-23.2.0/compute-envs/aws-batch.mdx:180]: Is this still in preview? If no, when did it graduate?

File: platform_versioned_docs/version-23.2.0/enterprise/configuration/overview.mdx [hunk 2]
```diff
@@ -44,7 +42,9 @@ Specify general Tower configuration values in your environment variables. The bo
 
 `TOWER_CRON_SERVER_PORT`: Define the HTTP port usd by the Tower cron service (default: `8080`, requires Tower 21.06.1 or later).
 
-`TOWER_USER_WORKSPACE_ENABLED` : Enable or disable the showing of the user private workspace context. (default: `true`, requires Tower 22.1.0 or later).
+`TOWER_USER_WORKSPACE_ENABLED`: Enable or disable the showing of the user private workspace context. (default: `true`, requires Tower 22.1.0 or later).
+
+`TOWER_ENABLE_ARM64`: Enable support for ARM64 CPU architectures. Required to enable Graviton in AWS Batch compute environments.
```
Comment (gwright99) [inline platform_versioned_docs/version-23.2.0/enterprise/configuration/overview.mdx:47]: I would qualify this with "Currently only available for ..." (probably exclusive to AWS Batch right now).

File: platform_versioned_docs/version-23.3.0/compute-envs/aws-batch.mdx [hunk 3]
```diff
@@ -161,7 +161,9 @@ Fargate requires the Fusion v2 file system and a **spot** provisioning model. Fa
 20. Select **Use Graviton CPU architecture** to execute on Graviton-based EC2 instances (ARM64 CPU architecture). When enabled, `c6g` instance types are used by default for compute jobs. You can specify your own **Instance types** under [**Advanced options**](#advanced-options). Seqera Platform supports all AWS Batch-compatible Graviton2 and Graviton3 instance types.
 
     :::note
-    Graviton requires Fargate, Wave containers, and Fusion v2 file system to be enabled. This feature is not compatible with GPU-based architecture.
+    Graviton requires Fargate, Wave containers, and Fusion v2 file system to be enabled. This feature is not compatible with GPU-based architecture. 
+    
+    In Tower Enterprise deployments, set `TOWER_ENABLE_ARM64=true` to enable Graviton instance support.
```
Comment (gwright99) [inline platform_versioned_docs/version-23.3.0/compute-envs/aws-batch.mdx:166]: ditto

File: platform_versioned_docs/version-23.3.0/enterprise/configuration/configtables/features_env.yml [hunk 4]
```diff
@@ -9,6 +9,11 @@
   Description: > 
     Define the Wave containers service endpoint URL. 
   Value:                'Example: `https://wave.seqera.io`'
+- 
+  Environment variable:            '`TOWER_ENABLE_ARM64`'
+  Description: > 
+    Enable support for ARM64 CPU architecture. Required to enable Graviton instance support in AWS Batch compute environments.
```
Comment (gwright99) [inline platform_versioned_docs/version-23.3.0/enterprise/configuration/configtables/features_env.yml:15]: This this feature 1:1 AWS Batch only? Or just happens that AWS Batch is the only ARM-type env you can activate right now (but expected that we can add other vendors in future)?

File: platform_versioned_docs/version-23.4.0/enterprise/release_notes/enterprise_latest.mdx [hunk 5]
```diff
@@ -59,6 +59,10 @@ If you specify a DNS name as your `TOWER_SERVER_URL`, but access your Seqera ins
 
 A previous iteration of the rebranded Seqera documentation noted `seqera` as the default and example value for certain variables (such as default database names). The rebranding from Nextflow Tower to Seqera Platform is an ongoing, incremental process and as such, legacy `tower` values and naming conventions used by the Seqera backend will remain in place until a future release. Updates to configuration variables and values will be communicated well in advance to prepare users for any breaking changes.
 
+**ARM64 CPU architecture support enabled by default**
+
+The `TOWER_ENABLE_ARM64` configuration environment variable is no longer needed to enable Graviton instance support in AWS Batch compute environments. ARM64 CPU architecture support is enabled by default.
```
Comment (gwright99) [inline platform_versioned_docs/version-23.4.0/enterprise/release_notes/enterprise_latest.mdx:64]: Qualify how this gets enabled? I assume this depends on the sort of machine selected? Some other field toggled to true?

**Review by gwright99 — CHANGES_REQUESTED** [review]
A few qualifications requested, please.

File: platform_versioned_docs/version-23.4.0/enterprise/release_notes/enterprise_latest.mdx (hunk 5 above)
Comment (llewellyn-sl) [reply platform_versioned_docs/version-23.4.0/enterprise/release_notes/enterprise_latest.mdx:64]: As in, the Graviton feature toggle in your AWS Batch CE form is now present by default. Previously, without the env var, you wouldn't even see the Graviton option in the form.

File: platform_versioned_docs/version-23.2.0/compute-envs/aws-batch.mdx (hunk 1 above)
Comment (llewellyn-sl) [reply platform_versioned_docs/version-23.2.0/compute-envs/aws-batch.mdx:180]: It was only preview for 23.2

File: platform_versioned_docs/version-23.4.0/enterprise/release_notes/enterprise_latest.mdx (hunk 5 above)
Comment (llewellyn-sl) [reply platform_versioned_docs/version-23.4.0/enterprise/release_notes/enterprise_latest.mdx:64]: Clarified entry, front-loading the Graviton feature toggle being available by default.

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
LGTM! Ready to merge!

**Pull request merged**